cli/src/cli_util.rs

```diff
@@ -1017,15 +1017,33 @@ impl WorkspaceCommandHelper {
                 let mut iter = [commit0, commit1].into_iter().chain(iter);
                 let commits: Vec<_> = iter.by_ref().take(5).try_collect()?;
                 let elided = iter.next().is_some();
-                let hint = format!(
-                    r#"The revset "{revision_str}" resolved to these revisions:{eol}{commits}{ellipsis}"#,
-                    eol = "\n",
-                    commits = commits
-                        .iter()
-                        .map(|c| self.format_commit_summary(c))
-                        .join("\n"),
-                    ellipsis = elided.then_some("\n...").unwrap_or_default()
-                );
+                let hint = if let RevsetExpression::CommitRef(RevsetCommitRef::Symbol(
+                    branch_name,
+                )) = revset_expression.as_ref()
+                {
+                    // Separate hint if there's a conflicted branch
+                    format!(
+                        r#"Branch {branch_name} resolved to multiple revisions because it's conflicted.
+It resolved to these revisions:
+{commits}{ellipsis}
```

Comment:
it looks like the `{commits}{ellipsis}` bit is the same between these two cases. Can you extract a variable for it?